mypy/util.py

```diff
@@ -467,43 +467,66 @@ class FancyFormatter:
-        self.BOLD = bold.decode()
-        self.UNDER = under.decode()
-        dim = curses.tigetstr('dim')
-        # TODO: more reliable way to get gray color good for both dark and light schemes.
-        self.DIM = dim.decode() if dim else PLAIN_ANSI_DIM
-
-        self.BLUE = curses.tparm(set_color, curses.COLOR_BLUE).decode()
-        self.GREEN = curses.tparm(set_color, curses.COLOR_GREEN).decode()
-        self.RED = curses.tparm(set_color, curses.COLOR_RED).decode()
-        self.YELLOW = curses.tparm(set_color, curses.COLOR_YELLOW).decode()
-
+        if sys.platform == 'win32':
+            # Windows ANSI escape sequences are only supported on Threshold 2 and above.
+            if sys.getwindowsversion().build < 10586:
+                self.dummy_term = True
+                return
+            self.dummy_term = False
+            import ctypes
+            kernel32 = ctypes.windll.kernel32
+            ENABLE_PROCESSED_OUTPUT = 0x1
+            ENABLE_WRAP_AT_EOL_OUTPUT = 0x2
+            ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x4
+            STD_OUTPUT_HANDLE = -11
+            kernel32.SetConsoleMode(kernel32.GetStdHandle(STD_OUTPUT_HANDLE),
+                                    ENABLE_PROCESSED_OUTPUT
+                                    | ENABLE_WRAP_AT_EOL_OUTPUT
+                                    | ENABLE_VIRTUAL_TERMINAL_PROCESSING)
+            self.BOLD = '\033[1m'
+            self.UNDER = '\033[4m'
+            self.BLUE = '\033[94m'
+            self.GREEN = '\033[92m'
+            self.RED = '\033[91m'
+            self.YELLOW = '\033[93m'
+            self.NORMAL = '\033[0m'
+            self.DIM = '\033[2m'
+        else:
+            # We in a human-facing terminal, check if it supports enough styling.
+            if not CURSES_ENABLED:
+                self.dummy_term = True
+                return
+            try:
+                curses.setupterm()
+            except curses.error:
+                # Most likely terminfo not found.
+                self.dummy_term = True
+                return
+            bold = curses.tigetstr('bold')
+            under = curses.tigetstr('smul')
+            set_color = curses.tigetstr('setaf')
+            self.dummy_term = not (bold and under and set_color)
+            if self.dummy_term:
+                return
+
+            self.NORMAL = curses.tigetstr('sgr0').decode()
+            self.BOLD = bold.decode()
+            self.UNDER = under.decode()
+            dim = curses.tigetstr('dim')
+            # TODO: more reliable way to get gray color good for both dark and light schemes.
+            self.DIM = dim.decode() if dim else PLAIN_ANSI_DIM
+
+            self.BLUE = curses.tparm(set_color, curses.COLOR_BLUE).decode()
+            self.GREEN = curses.tparm(set_color, curses.COLOR_GREEN).decode()
+            self.RED = curses.tparm(set_color, curses.COLOR_RED).decode()
+            self.YELLOW = curses.tparm(set_color, curses.COLOR_YELLOW).decode()
```

Comment:
An idea for refactoring: move the blocks inside `if` and `else` to two methods: `initialize_win_colors()` and `initialize_unix_colors()`?